probe-rs/src/core/mod.rs

```diff
@@ -509,6 +509,10 @@ impl Core {
         }
     }
 
+    pub fn architecture(&self) -> Architecture {
```

Comment:
I would appreciate a doc comment (even if it's kind of obvious what it does) on all further public functions (we should annotate all of them, but if we do it from now on onwards, it makes it easier).